src/lib/simulator/Simulator.h

```diff
@@ -45,163 +46,158 @@ class Simulator {
    * \param particles the particle container
    * \param plotter An instance plotter.
    * \param config the runtime configuration
-   * */
-  explicit Simulator(container::ParticleContainer<DIMENSIONS> &&particles, physics::ForceModel physics, std::unique_ptr<io::Plotter<DIMENSIONS>> &&plotter, const std::shared_ptr<config::Config> &config, Checkpointer<DIMENSIONS> checkpoint)
+   * \param checkpoint the checkpointer instance
+   */
+  explicit Simulator(std::unique_ptr<container::Container<DIMENSIONS>> &&particles,
+                     std::unique_ptr<io::Plotter<DIMENSIONS>> &&plotter,
+                     const std::shared_ptr<config::Config> &config,
+                     Checkpointer<DIMENSIONS> checkpoint,
+                     std::optional<simulator::physics::MembraneForce> membrane_force)
       : particles(std::move(particles)),
-        physics(physics),
         plotter(std::move(plotter)),
         config(config),
         thermostat(config->temp_init, config->temp_target, config->max_temp_diff, config->seed),
         checkpoint(checkpoint),
+        profile_calculator(config->profile_bins),
+        membrane_force(membrane_force),
         end_time(config->end_time),
         delta_t(config->delta_t),
         gravity(config->ljf_gravity) {};
 
+  /**
+   * \brief Calculates the new position of a particle.
+   * \param p The particle container.
+   * \param index The index of the particle.
+   */
   auto inline calculate_position_particle(Particles<DIMENSIONS> &p, size_t index) const {
-
     const auto temp = pow(delta_t, 2) * (1 / (2 * p.mass[index]));
-    for (int i = 0; i < DIMENSIONS; ++i) {
-      p.positions[i][index] += delta_t * p.velocities[i][index] + temp * p.old_forces[i][index];
+    if (p.fixed[index] == 0) [[likely]] {
+      for (size_t i = 0; i < DIMENSIONS; ++i) {
+        p.positions[i][index] += delta_t * p.velocities[i][index] + temp * p.old_forces[i][index];

```

Comment:
We updated the slides for next semester since you are not the only group with this mistake.
But you should use the updated forces here, not the old ones.